packages/camera/ios/Classes/CameraPlugin.m

```diff
@@ -0,0 +1,290 @@
+      [AVCaptureConnection connectionWithInputPorts:input.ports output:output];
+  if ([_captureDevice position] == AVCaptureDevicePositionFront) {
+    connection.videoMirrored = YES;
+  }
+  connection.videoOrientation = AVCaptureVideoOrientationPortrait;
+  [_captureSession addInputWithNoConnections:input];
+  [_captureSession addOutputWithNoConnections:output];
+  [_captureSession addConnection:connection];
+  _capturePhotoOutput = [AVCapturePhotoOutput new];
+  [_captureSession addOutput:_capturePhotoOutput];
+  return self;
+}
+
+- (void)start {
+  [_captureSession startRunning];
+}
+
+- (void)stop {
+  [_captureSession stopRunning];
+}
+
+- (void)captureToFile:(NSString *)path result:(FlutterResult)result {
+  AVCapturePhotoSettings *settings = [AVCapturePhotoSettings photoSettings];
+  [_capturePhotoOutput
+      capturePhotoWithSettings:settings
+                      delegate:[[FLTSavePhotoDelegate alloc] initWithPath:path result:result]];
+}
+
+- (void)captureOutput:(AVCaptureOutput *)output
+    didOutputSampleBuffer:(CMSampleBufferRef)sampleBuffer
+           fromConnection:(AVCaptureConnection *)connection {
+  CVPixelBufferRef newBuffer = CMSampleBufferGetImageBuffer(sampleBuffer);
+  CFRetain(newBuffer);
+  CVPixelBufferRef old = _latestPixelBuffer;
+  while (!OSAtomicCompareAndSwapPtrBarrier(old, newBuffer, (void **)&_latestPixelBuffer)) {
+    old = _latestPixelBuffer;
+  }
+  if (old != nil) {
+    CFRelease(old);
+  }
+  if (_onFrameAvailable) {
+    _onFrameAvailable();
+  }
+}
+
+- (void)close {
+  [_captureSession stopRunning];
+  for (AVCaptureInput *input in [_captureSession inputs]) {
+    [_captureSession removeInput:input];
+  }
+  for (AVCaptureOutput *output in [_captureSession outputs]) {
+    [_captureSession removeOutput:output];
+  }
+}
+
+- (void)dealloc {
+  if (_latestPixelBuffer) {
+    CFRelease(_latestPixelBuffer);
+  }
+}
+
+- (CVPixelBufferRef)copyPixelBuffer {
+  CVPixelBufferRef pixelBuffer = _latestPixelBuffer;
+  while (!OSAtomicCompareAndSwapPtrBarrier(pixelBuffer, nil, (void **)&_latestPixelBuffer)) {
+    pixelBuffer = _latestPixelBuffer;
+  }
+  return pixelBuffer;
+}
+
+- (FlutterError *_Nullable)onCancelWithArguments:(id _Nullable)arguments {
+  _eventSink = nil;
+  return nil;
+}
+
+- (FlutterError *_Nullable)onListenWithArguments:(id _Nullable)arguments
+                                       eventSink:(nonnull FlutterEventSink)events {
+  _eventSink = events;
+  return nil;
+}
+@end
+
+@interface CameraPlugin ()
+@property(readonly, nonatomic) NSObject<FlutterTextureRegistry> *registry;
+@property(readonly, nonatomic) NSObject<FlutterBinaryMessenger> *messenger;
+@property(readonly, nonatomic) NSMutableDictionary *cams;
+@end
+
+@implementation CameraPlugin
++ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar> *)registrar {
+  FlutterMethodChannel *channel =
+      [FlutterMethodChannel methodChannelWithName:@"camera" binaryMessenger:[registrar messenger]];
```

Comment:
Channel name should have a namespace prefix, `plugins.flutter.io/camera`.

Reply:
Oops.
Done